web/modules/custom/dpl_filter_paragraphs/dpl_filter_paragraphs.module

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * @file
+ * DPL filtered Paragraphs.
+ */
+
+use Drupal\Component\Utility\NestedArray;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\views\Views;
+
+/**
+ * Implements theme_preprocess_paragraph__NAME().
+ *
+ * @see _dpl_filter_paragraphs_prepare_filter_view()
+ */
+function dpl_filter_paragraphs_preprocess_paragraph__card_grid_automatic(array &$variables): void {
+  $variables = _dpl_filter_paragraphs_prepare_filter_view($variables, 'card_grid');
+}
+
+/**
+ * Getting the content list views, and setting relevant paragraph filters.
+ *
+ * We use the filters to set as contextual filters on the view.
+ *
+ * @param array<mixed> $variables
+ *   See the $variables in theme_preprocess_paragraph__NAME().
+ *
+ * @return array<mixed>
+ *   See the $variables in theme_preprocess_paragraph__NAME().
+ */
+function _dpl_filter_paragraphs_prepare_filter_view(array $variables, string $item_view_mode): array {
+  // In the preview (AKA backend view), we don't want to load in the whole view.
+  if ($variables['view_mode'] === 'preview') {
+    return $variables;
+  }
+
+  $variables['#cache']['tags'][] = 'node_list';
```

Comment:
I wonder why this is necessary. I would expect the child element with the view to have the necessary cache tags including `node_list` and for these to bubble up. 🤔

Reply:
I'll add a code comment to clarifiy :)